tests/Integration/Behaviour/Features/Context/ShopFeatureContext.php

```diff
@@ -49,19 +53,138 @@ public function shopWithNameExists(string $reference, string $shopName)
+        Db::getInstance()->execute(
+            'INSERT INTO `' . _DB_PREFIX_ . 'country_shop` (`id_country`, `id_shop`) ' .
+            'SELECT id_country, ' . $shop->id . ' ' .
+            'FROM `' . _DB_PREFIX_ . 'country` '
+        );
+
+        SharedStorage::getStorage()->set($reference, $shop);
+    }
+
     /**
      * @Given single shop context is loaded
      */
     public function singleShopContextIsLoaded()
     {
-        Shop::setContext(Shop::CONTEXT_SHOP, Configuration::get('PS_SHOP_DEFAULT'));
+        $this->setShopContext(Shop::CONTEXT_SHOP, (int) Configuration::get('PS_SHOP_DEFAULT'));
     }
 
     /**
      * @Given multiple shop context is loaded
      */
     public function multipleShopContextIsLoaded()
     {
-        Shop::setContext(Shop::CONTEXT_ALL);
+        $this->setShopContext(Shop::CONTEXT_ALL, (int) Configuration::get('PS_SHOP_DEFAULT'));
+    }
+
+    /**
+     * @Given /^shop context "(.+)" is loaded$/
+     *
+     * @param string $shopName
+     */
+    public function specificShopContextIsLoaded(string $shopName): void
+    {
+        $this->setShopContext(Shop::CONTEXT_SHOP, (int) Shop::getIdByName($shopName));
+    }
+
+    /**
+     * @Then /^I should have (\d) shop group(s)$/
+     *
+     * @param int $expectedCount
+     */
+    public function checkShopGroupCount(int $expectedCount)
+    {
+        $countShopGroup = ShopGroup::getTotalShopGroup();
+
+        if ($countShopGroup == $expectedCount) {
+            return;
+        }
+        throw new RuntimeException(
+            sprintf(
+                'Invalid number of shop groups, expected %s but got %s instead',
+                $expectedCount,
+                $countShopGroup
+            )
+        );
+    }
+
+    /**
+     * @Then /^I should have (\d) shop(?:|s) in group "(.+)"$/
+     *
+     * @param int $expectedCount
+     */
+    public function checkShopCount(int $expectedCount, string $shopGroupName)
+    {
+        $shopGroupId = ShopGroup::getIdByName($shopGroupName);
+        if (false === $shopGroupId) {
+            throw new RuntimeException(sprintf('Shop Group with name "%s" does not exist', $shopGroupName));
+        }
+
+        $shops = ShopGroup::getShopsFromGroup($shopGroupId);
+        if (count($shops) == $expectedCount) {
+            return;
+        }
+        throw new RuntimeException(
+            sprintf(
+                'Invalid number of shop groups, expected %s but got %s instead',
+                $expectedCount,
+                count($shops)
+            )
+        );
+    }
+
+    /**
+     * @param int $context
+     * @param int $shopId
+     *
+     * @throws \PrestaShopException
+     */
+    private function setShopContext(int $context, int $shopId): void
+    {
+        Shop::setContext($context, $shopId);
+        Context::getContext()->shop = new Shop($shopId);
+        // Clean cache
+        Cache::clean('Shop::getCompleteListOfShopsID');
+        Cache::clean('StockAvailable::*');
```

Comment:
I see so many functions dedicated to Cache, should we group them all together in some "Util class" ? We can use helper classes in Behat tests.

Reply:
It's not in the PR scope.